### DISCUSSION TITLE
Add a "Favourite" or "Follow" ability for packages or scopes

### DISCUSSION THREAD

**Discussion by NuroDev** [jsr-io/jsr#792] [Ideas]
As a nice little quality-of-life feature it would be cool to offer an ability to "favourite" packages and/or "follow" scopes so you can track your most liked packages.

Example: I love both the [`@std`](https://jsr.io/@std) & [`@cross`](https://jsr.io/@cross) scopes in JSR. I would love to be able to "favourite" or "follow" them & go to a page to view these packages / scopes in one place.

This could also potentially include sub-categories in the future so I could "favourite" a bunch of my favourite testing utility packages and all kept under that one sub-category.

**Comment by SrGeneroso** [comment]
Oh, dang... I just opened a new discussion on this... I even clicked the "I've search for similar content" before posting it!
Let me copy it here and close that one.

**Comment by SrGeneroso** [comment]
Hi, I want to propose a feature for JSR.io
I might have a problem and I feel the necessity of bookmark and categorize everything I like. I would love for jsr.io to have a "Like/Star" function to save packages for later. Also, it would be great if those could be categorize on Lists of some sort. This feature is present in github and youtube.
Proof of my need for saving things on those platforms.

![Image](https://github.com/user-attachments/assets/3a8f199e-ba2d-4d77-ba87-bf2d745be7e2)

Github has stars and you can categorize them in lists (Did you know you can only have 32 different ones)

![Image](https://github.com/user-attachments/assets/b7807b9b-5cb4-4dae-b159-4a0ca8951729)

And youtube let you save into playlists
![Image](https://github.com/user-attachments/assets/02b4097c-3116-44de-b291-72d61a19ee6d)

![Image](https://github.com/user-attachments/assets/c13626b8-e305-4514-8c3b-ee2558afdf1e)

Please note that I always prefix the different lists with emojis and I'm not ashamed for it.

For JSR.io, I propose a similar approach to pin packages and use list for things like: "🔐auth" - "📅date" - "🗺️geo" - "🎨UI" - "📑validation"... (my lists could go on forever...) or maybe based in specific projects (a list for my own crm, or for svelte projects...) made by the users for personal organization and to share with others. I think they should be able to be private or public, since people may don't want to make public which packages they use for security or whatever reason.

Another use of this lists, could be to create recipes for starter projects or a pick and choose with maybe a sub-feature to generate the code to install those packages
```bash
deno add jsr:@std/internal jsr:@std/crypto jsr:@std/testing
```
like you get with individual packages
![Image](https://github.com/user-attachments/assets/116e9149-0be2-4868-b53c-75486553f253)
(darkmode please 😎)

I guess at this moment only publishers of packages have or need accounts, so I'm not sure if this feature could be too costly but for sure would increase the chances of new users.

For UI, I guess we could have a new tab in the user profile

![Image](https://github.com/user-attachments/assets/87248768-65c0-4e70-adaf-dc94c6c85bb9)

and it could have a button to save it in each package with a modal like google or github does.

![Image](https://github.com/user-attachments/assets/af0162be-90ff-4af8-a95b-581f74ae732d)
Please note that my UI design is quite poor and could be very much improved.

In any case, I don't have a proper spec list for this feature but I think it could be cool. I know platforms like github, which *shouldn't be considered social media have this kind of features and don't get a lot of love, in fact what you find a lot are [awesome list repos](https://github.com/denolib/awesome-deno) and I find that npm should and JSR could have this feature.

I had this originally posted in a [thread](https://discord.com/channels/1308846349051232306/1345791837561950249) in the discord server. Go there if you want to chat about it more informally.

**Comment by EGAMAGZ** [comment]
The "save package" feature, which allows you to keep specific packages in the loop, is a useful addition and I like it. However, both this and the following feature may deviate from the core purpose of a package registry, such as [Pypi](https://pypi.org/), [NPM](https://www.npmjs.com/), [Crates](https://crates.io/), or [hex.pm](https://hex.pm/).

**Reply by AugustinMauroy** [reply]
agree with you. With JSR api, I think it's easy to make your own app that solves this problem.

**Reply by SrGeneroso** [reply]
I understand the core purpose of a package registry is far from this "social network" feature, I guess it was as well for github original purpose, and apart for (in the github example) showing some love to the projects starring them, I don't think many people use the lists or search in their starred repos when looking for projects they've seen in the past.
I feel like I'm on the side of the few users that even create playlists in youtube or github, I'm a hoarder and I should look for help (the new obsidian extension is not helping me in that front).
As you guys mentioned, Is not hard to make my own solution for that, simply bookmarks or markdown list are more than enough and would alleviate resources in the jsr database of users, although, it could also give a lot of insight into what users like or use. I'm in the fence about tracking users and exploiting data, but I guess today consumers of jsr, npm or other registries don't have accounts for any of them, you only need it if you are a publisher. Features like the one I propose could evolve into more consumer engaging with the platform outside of the original purpose, but since most projects are hosted on github I guess I could go there instead.

**Comment by AugustinMauroy** [comment]
About follow, we can generate RSS feed on package/scope.
